imap_data_access/processing_input.py

```diff
@@ -325,22 +386,27 @@ def _set_attributes_from_filenames(self) -> None:
                 self.descriptor = "best"
 
         if "spin" in source:
-            # Update the source to be spin
+            # Update the source and data type to be spin
             self.source = SPICESource.SPIN.value
+            self.data_type = SPICESource.SPIN.value
             if len(source) != 1:
                 raise ValueError(
                     "If spin data in the list, it should only contain spin files"
                 )
         elif "repoint" in source:
-            # Update the source to be repoint
+            # Update the source and data type to be repoint
             self.source = SPICESource.REPOINT.value
+            self.data_type = SPICESource.REPOINT.value
             # Latest file will contain all the repointing data.
             if len(file_obj_list) != 1:
                 raise ValueError(
                     "There should only be one repoint file in the list of files"
                 )
         else:
-            self.source = SPICESource.SPICE.value
+            # Update the source and data type to be file type and spice, respectively
+            # Join all the sources into a single string separated by commas
+            self.source = ",".join(source)
```

Comment:
Why do we convert these to strings, could we keep it as a list instead? I feel like the to/from string should be done on casting to the call sites that need strings, but otherwise it seems nicer as a list of items internally.